java/sidecar-common/src/main/java/com/instaclustr/sidecar/operations/OperationsFilter.java

```diff
@@ -0,0 +1,35 @@
+package com.instaclustr.sidecar.operations;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.base.MoreObjects;
+import com.instaclustr.sidecar.jackson.OperationTypeToStringConverter;
+import com.instaclustr.sidecar.jackson.StringToOperationTypeConverter;
+import com.instaclustr.sidecar.validation.Enum;
+
+public class OperationsFilter {
+
+    @JsonSerialize(converter = OperationTypeToStringConverter.class)
+    @JsonDeserialize(converter = StringToOperationTypeConverter.class)
+    public final OperationType type;
+
+    @Enum(enumClass = Operation.State.class, ignoreCase = true)
+    public final String state;
```

Comment:
Here it is `string` in order to put Enum constrain on that so we will have fully under control the returning message when user does not specify one of states, like this:

    [
        {
            "message": "Value complet is invalid for enumeration class com.instaclustr.sidecar.operations.Operation$State. Possible values: PENDING,RUNNING,COMPLETED,FAILED",
            "messageTemplate": "Value complet is invalid for enumeration class com.instaclustr.sidecar.operations.Operation$State. Possible values: PENDING,RUNNING,COMPLETED,FAILED",
            "path": "OperationsResource.operations.arg0.state",
            "invalidValue": "complet"
        }
    ]

If it was just `State`, it would say that there is not such enum / it could not be deserialised, but we would not know what values are possible and so on.

Reply:
outdated